### DISCUSSION TITLE
Can I add Android, IOS, and Linux libraries?

### DISCUSSION THREAD

**Discussion by long1752062104** [recastnavigation/recastnavigation#663] [Q&A] [answered]
Mainly used in Unity, between servers and clients. However, due to my limited abilities, I am unable to encapsulate the libraries of these platforms. I hope the author can encapsulate them for me. Thank you

**Comment by grahamboree** [comment, marked as answer]
This is a frequent request and is something that's on our [roadmap](https://github.com/recastnavigation/recastnavigation/blob/main/Roadmap.md).  

The long and short of it is that Recast is currently a C++ only library and makes some light use of objects as part of its mechanisms.  This makes it difficult (but not impossible) to provide a nice C interface, and therefore provide usable pre-built libraries.

My hope is that we can move the core of the library from C++ to C and provide an official C API.  That would make this trivial and it would add better official support for other languages through their C FFI.
